app/parsers/katz.py

```diff
@@ -0,0 +1,124 @@
+            package = {}
+            package_name = ''
+            in_header = True
+            for line in section.splitlines():
+                line = line.strip()
+                if in_header:
+                    if line.startswith('msv'):
+                        in_header = False
+                    else:
+                        session = re.match(r'^\s*Session\s*:\s*([^\r\n]*)', line)
+                        if session:
+                            mk_section.session = session.group(1)
+                        username = re.match(r'^\s*User Name\s*:\s*([^\r\n]*)', line)
+                        if username:
+                            mk_section.username = username.group(1)
+                        domain = re.match(r'^\s*Domain\s*:\s*([^\r\n]*)', line)
+                        if domain:
+                            mk_section.domain = domain.group(1)
+                        logon_server = re.match(r'^\s*Logon Server\s*:\s*([^\r\n]*)', line)
+                        if logon_server:
+                            mk_section.logon_server = logon_server.group(1)
+                        logon_time = re.match(r'^\s*Logon Time\s*:\s*([^\r\n]*)', line)
+                        if logon_time:
+                            mk_section.logon_time = logon_time.group(1)
+                        sid = re.match(r'^\s*SID\s*:\s*([^\r\n]*)', line)
+                        if sid:
+                            mk_section.sid = sid.group(1)
+                        continue
+
+                if line.startswith('['):
+                    # this might indicate the start of a new account
+                    if 'Username' in package and package['Username'] != '(null)' and \
+                            (('Password' in package and package['Password'] != '(null)') or 'NTLM' in package):
+                        mk_section.packages[package_name].append(package)
+
+                    # reset the package
+                    package = {}
+                    pass
+                elif line.startswith('*'):
+                    m = re.match(r'\s*\* (.*?)\s*: (.*)', line)
+                    if m:
+                        package[m.group(1)] = m.group(2)
+
+                elif line:
+                    # parse out the new section name
+                    match_group = re.match(r'([a-z]+) :', line)
+                    if match_group:
+                        # this is the start of a new ssp
+                        # save the current ssp if necessary
+                        if 'Username' in package and package['Username'] != '(null)' and \
+                                (('Password' in package and package['Password'] != '(null)') or 'NTLM' in package):
+                            mk_section.packages[package_name].append(package)
+
+                        # reset the package
+                        package = {}
+
+                        # get the new name
+                        package_name = match_group.group(1)
```

Comment:
private function?

Reply:
This one looks less likely because you're resetting the package, but still worth considering

Reply:
I think I found a way by bundling it with a few other things.